index.js

```diff
@@ -38,153 +39,179 @@ module.exports.s3 = function(config, key, args, callback) {
-							return Number(value.substr(0, value.length - 2));
+
+					// auto rotate based on orientation exif data
+					image.rotate();
+
+					// convert gifs to pngs unless animated
+					if (
+						args.key &&
+						path.extname(args.key).toLowerCase() === '.gif'
+					) {
+						if (isAnimated(buffer)) {
+							return callback(new Error('fallback-to-original'));
 						} else {
-							return Number(
-								Number(
-									metadata[index % 2 ? 'height' : 'width'] *
-										(value / 100)
-								).toFixed(0)
-							);
+							image.png();
 						}
-					});
-
-					image.extract({
-						left: cropValues[0],
-						top: cropValues[1],
-						width: cropValues[2],
-						height: cropValues[3],
-					});
-				}
-
-				// resize
-				if (args.resize) {
-					args.resize =
-						typeof args.resize === 'string'
-							? args.resize.split(',')
-							: args.resize;
-					image.resize.apply(
-						image,
-						args.resize.map(function(v) {
-							return Number(v) || null;
-						})
-					);
-					if ( args.gravity ) {
-						image.crop( args.gravity );
 					}
-				} else if (args.fit) {
-					args.fit =
-						typeof args.fit === 'string'
-							? args.fit.split(',')
-							: args.fit;
-					image.resize.apply(
-						image,
-						args.fit.map(function(v) {
-							return Number(v) || null;
-						})
-					);
-					image.max();
-				} else if (args.lb) {
-					args.lb =
-						typeof args.lb === 'string'
-							? args.lb.split(',')
-							: args.lb;
-					image.resize.apply(
-						image,
-						args.lb.map(function(v) {
-							return Number(v) || null;
-						})
-					);
-
-					// default to a black background to replicate Photon API behaviour
-					// when no background colour specified
-					if (!args.background) {
-						args.background = 'black';
+
+					// crop (assumes crop data from original)
+					if (args.crop && !args.crop.match(/(smart|attention|entropy)/)) {
+						var cropValues =
+							typeof args.crop === 'string'
+								? args.crop.split(',')
+								: args.crop;
+
+						// convert percentages to px values
+						cropValues = cropValues.map(function(value, index) {
+							if (value.indexOf('px') > -1) {
+								return Number(value.substr(0, value.length - 2));
+							} else {
+								return Number(
+									Number(
+										metadata[index % 2 ? 'height' : 'width'] *
+											(value / 100)
+									).toFixed(0)
+								);
+							}
+						});
+
+						image.extract({
+							left: cropValues[0],
+							top: cropValues[1],
+							width: cropValues[2],
+							height: cropValues[3],
+						});
 					}
-					image.background(args.background);
-					image.embed();
-				} else if (args.w || args.h) {
-					image.resize(
-						Number(args.w) || null,
-						Number(args.h) || null
-					);
-					if (!args.crop) {
+
+					// resize
+					if (args.resize) {
+						args.resize = getDimArray( args.resize );
+
+						// apply cropping strategies
+						if ( args.crop === 'attention' ) {
```

Comment:
Seems strange to use the `resize` param for cropping? Right now, so far as I can see, they're two independent operations, so you could e.g. crop an image down to a 200x200 square then resize to 64x64.

Reply:
Resizing doesn't preserve aspect ratio so causes cropping by default in sharp. It's the param we use in the WP plugin to generate image sizes that have `crop` set to true so I wasn't really thinking of it that way.

I think it's much harder to figure out a crop size and then resize personally.